### PR TITLE
Feature/improve custom types support

### PR DESCRIPTION
**Pull request by ibarbashov**

### CODE REVIEW

File: oracle-plugin/src/main/java/co/cask/oracle/OracleSchemaReader.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    if (ORACLE_TYPES.contains(sqlType)) {
+      switch (sqlType) {
+        case INTERVAL_DS:
+        case INTERVAL_YM:
+          return Schema.of(Schema.Type.STRING);
+        case TIMESTAMP_LTZ:
+        case TIMESTAMP_TZ:
+          return Schema.of(Schema.LogicalType.TIMESTAMP_MICROS);
+        default:
+          throw new SQLException(new UnsupportedTypeException("Unsupported SQL Type: " + sqlType));
```
Comment (aielchiieva) [inline oracle-plugin/src/main/java/co/cask/oracle/OracleSchemaReader.java]: Does `metadata` have column name? If yes, please add it to the exception message.

File: postgresql-plugin/src/main/java/co/cask/postgres/PostgresSchemaReader.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+/**
+ * PostgreSQL schema reader.
+ */
+public class PostgresSchemaReader extends CommonSchemaReader {
+
+  public static final Set<Integer> POSTGRES_TYPES = ImmutableSet.of(
+    Types.OTHER, Types.ARRAY, Types.SQLXML
+  );
+
+  public PostgresSchemaReader() {
```
Comment (aielchiieva) [inline postgresql-plugin/src/main/java/co/cask/postgres/PostgresSchemaReader.java]: Seems that defining default constructor is unnecessary.

File: postgresql-plugin/src/main/java/co/cask/postgres/PostrgesDBRecord.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+/**
+ * Writable class for PostgreSQL Source/Sink
+ */
+public class PostrgesDBRecord extends DBRecord {
+
+  public PostrgesDBRecord(StructuredRecord record, int[] columnTypes) {
+    this.record = record;
```
Comment (aielchiieva) [inline postgresql-plugin/src/main/java/co/cask/postgres/PostrgesDBRecord.java]: Use super constructor.

File: postgresql-plugin/src/main/java/co/cask/postgres/PostrgesDBRecord.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+   * Used in map-reduce. Do not remove.
+   */
+  @SuppressWarnings("unused")
+  public PostrgesDBRecord() {}
+
+  @Override
+  protected void handleField(ResultSet resultSet, StructuredRecord.Builder recordBuilder, Schema.Field field,
+                             int sqlType, int sqlPrecision, int sqlScale) throws SQLException {
+    if (PostgresSchemaReader.POSTGRES_TYPES.contains(sqlType)) {
+      handleOracleSpecificType(resultSet, recordBuilder, field, sqlType);
```
Comment (aielchiieva) [inline postgresql-plugin/src/main/java/co/cask/postgres/PostrgesDBRecord.java]: Why Oracle?

**Review by aielchiieva — COMMENTED** [review]
Also please add unit tests for Postgres.

File: oracle-plugin/src/main/java/co/cask/oracle/OracleSchemaReader.java (hunk 1 above)
Comment (ibarbashov) [reply oracle-plugin/src/main/java/co/cask/oracle/OracleSchemaReader.java]: done

File: postgresql-plugin/src/main/java/co/cask/postgres/PostgresSchemaReader.java (hunk 2 above)
Comment (ibarbashov) [reply postgresql-plugin/src/main/java/co/cask/postgres/PostgresSchemaReader.java]: done

File: postgresql-plugin/src/main/java/co/cask/postgres/PostrgesDBRecord.java (hunk 3 above)
Comment (ibarbashov) [reply postgresql-plugin/src/main/java/co/cask/postgres/PostrgesDBRecord.java]: done

File: postgresql-plugin/src/main/java/co/cask/postgres/PostrgesDBRecord.java (hunk 4 above)
Comment (ibarbashov) [reply postgresql-plugin/src/main/java/co/cask/postgres/PostrgesDBRecord.java]: fixed

**Comment by aielchiieva** [comment]
+1

**Pull request merged**